### ISSUE TITLE
BUG: When updating a vue data value, vue-instantsearch throws an error and stops working.

### ISSUE THREAD

**Issue by danielmarkavis** [algolia/vue-instantsearch#738]
**Bug 🐞**

### What is the current behavior?

TLDR: When updating a vue data value, vue-instantsearch throws an error and stops working.

`[InstantSearch.js]: The index named “” is not listed in the 'items' of 'sortBy'`

I wanted to create a button that slid in the menu element. But when I update the “slideIn” value in vue, it crashes the search, and receives the indexName error (see below).

It’s something to do with the routing system, and ais-sort?

### Make a sandbox with the current behavior

[Codepen Vue](https://codesandbox.io/s/vue-instantsearch-app-twejh)

### What is the expected behavior?
To not crash the search

### Does this happen only in specific situations?
When a value is update in vue.

### What is the proposed solution?
N/A

### What is the version you are using?
"algoliasearch": "^3.35.1",
"vue-instantsearch": "^2.6.0",

**Comment by Haroenv** [comment]
Unfortunately this is also #575. You can fix this by putting _no_ objects or arrays directly as props. It's a known limitation of Vue.

Here's a fixed sandbox: https://codesandbox.io/s/vue-instantsearch-app-dmhof

I'll close this, since the other issue lists actionable points.
